src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,923 @@
+	FindDashboardAPI    string   `json:"findDashboardAPI"`    // API endpoint for all dashboards with given tags
+	ListAnnotationsAPI  string   `json:"listAnnotationsAPI"`  // API endpoint for fetching all annotations
+	CreateAnnotationAPI string   `json:"createAnnotationAPI"` // API endpoint for creating an annotation
+	UpdateAnnotationAPI string   `json:"updateAnnotationAPI"` // API endpoint for updating an annotation
+	DeleteAnnotationAPI string   `json:"deleteAnnotationAPI"` // API endpoint for deleting an annotation
+	Columns             []string `json:"columns"`             // column names for annotations info
+	HTTPTimeout         int      `json:"httpTimeout"`         // http timeout to connect to AM
+	Verbose             int      `json:"verbose"`             // verbosity level
+	Token               string   `json:"token"`               // CERN SSO token to use
+}
+
+// -------STRUCTS---------
+
+// function for get request grafana annotations endpoint for fetching annotations.
+func getAnnotations(data *[]annotationData, tags []string) error {
+
+	var headers [][]string
+	bearer := fmt.Sprintf("Bearer %s", configJSON.Token)
+	h := []string{"Authorization", bearer}
+	headers = append(headers, h)
+	h = []string{"Accept", "application/json"}
+	headers = append(headers, h)
+	// Ex: /api/annotations?from=1598366184000&to=1598366184000&tags=cmsweb&tags=cmsmon-int
+	v := url.Values{}
+
+	if trange != "" {
+		tms := parseTimes(false) // False because Listing annotations with past time offset
+
+		if len(tms) == 2 {
+			v.Add("from", strings.Trim(strconv.FormatInt(tms[0], 10), " "))
+			v.Add("to", strings.Trim(strconv.FormatInt(tms[1], 10), " "))
+		} else if len(tms) == 1 {
+			v.Add("from", strings.Trim(strconv.FormatInt(tms[0], 10), " "))
+			v.Add("to", strings.Trim(strconv.FormatInt(time.Now().UTC().Unix()*int64(milliSec), 10), " "))
+		}
+	}
+
+	for _, tag := range tags {
+		v.Add("tags", strings.Trim(tag, " "))
+	}
+
+	apiURL := fmt.Sprintf("%s%s?%s", configJSON.GrafanaBaseURL, configJSON.ListAnnotationsAPI, v.Encode())
+
+	if configJSON.Verbose > 0 {
+		log.Println(apiURL)
+	}
+	req, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		log.Printf("Unable to make request to %s, error: %s", apiURL, err)
+		return err
+	}
+	for _, v := range headers {
+		if len(v) == 2 {
+			req.Header.Add(v[0], v[1])
+		}
+	}
+
+	timeout := time.Duration(configJSON.HTTPTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+
+	if configJSON.Verbose > 1 {
+		log.Println("URL", apiURL)
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("Request: ", string(dump))
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		log.Printf("Unable to read JSON Data from Grafana Annotation GET API, error: %v\n", err)
```

Comment:
replace with Fatal and adjust code which uses function. It seems to me there is no need to return an error if all errors will be handled within this function body.

Reply:
Fixed